include/zephyr/bluetooth/bluetooth.h

```diff
@@ -1816,6 +1821,8 @@ int bt_le_per_adv_sync_delete(struct bt_le_per_adv_sync *per_adv_sync);
  * This callback will be called for all periodic advertising sync activity,
  * such as synced, terminated and when data is received.
  *
+ * @note Registering the same callback struct twice is not supported.
+ *
```

Comment:
Actually it's worse than that. If the same callback is registered twice, then we end up in a situation where they callback in the slist point to itself, and we end in an infinite loop :(

Reply:
Yes, this is the issue we should solve :)

Reply:
Maybe we should phrase it a bit different? The implementation should probably trigger an assert or simply ignore the second registration

Reply:
> The implementation should probably trigger an assert or simply ignore the second registration

Indeed. I created https://github.com/zephyrproject-rtos/zephyr/issues/65973 a while ago to be able to fix this, and can be fixed by using the `sys_slist_find` from https://github.com/zephyrproject-rtos/zephyr/pull/66621 before appending to a list. 

Should we create a GH issue for it, or just fix it immediately?